### DISCUSSION TITLE
Enhancing Expo-Video API

### DISCUSSION THREAD

**Discussion by Sammy087** [expo/expo#27524] [Proposals and Ideas]
Hi, i'd like to enhance the cache and preloading of expo video as well as cache the DRM Keys. I want to create an API that has these features, and can be integrated seamlessly with Expo Video. I do have Gist proposals for how it could be done here. Would love some input on how we can go about getting a green light to do this! 
https://gist.github.com/HenriqueFadoni/015a97bd92f8ca1f2385f41f1ce41afe

Thanks

**Comment by Sammy087** [comment]
hello, still wondering about this!

**Comment by tsapeta** [comment]
Hey!
To be honest, after reading some of these options I'm not quite sure what you're trying to achieve.
We're no longer accepting any new features to `expo-av` as we're writing new `expo-video` and `expo-audio` from scratch, with completely new APIs.
I would love to know what's your use case, maybe we could cover that in these new packages.

**Reply by Sammy087** [reply]
Hey, Thanks for the response! 

So I just need more control over video pre-loading and cache management. We currently use expo-av, which as we know is being replace by expo-video. We're using hls on iOS and dash-mpeg video formats (with DRM protection).The video is all working fine, but I'd like a way to load up the first few seconds of a video without having to mount the video component and play it from the cache. Would also like to prefetch video as a result of a push notification or background process, but also a way to expire and remove these videos. I'd also like the DRM keys cached. The task from there is to unify the our API with your Video API into a common one and expose them in Javascript/Typescript.  A robust proposal would show how the Typescript API maps to the two different native layers. Would also be nice to expose things like priority for downloading and eviction (which are available on iOS, IDK about download manager though). How can we go about doing this?

**Reply by Sammy087** [reply]
our API would solve all the issues we have and want to change* That is was im preposing

**Reply by Sammy087** [reply]
Hey, just following up on this! Thanks!

**Comment by Sammy087** [comment]
> Hey! To be honest, after reading some of these options I'm not quite sure what you're trying to achieve. We're no longer accepting any new features to `expo-av` as we're writing new `expo-video` and `expo-audio` from scratch, with completely new APIs. I would love to know what's your use case, maybe we could cover that in these new packages.

hey, just wondering what you think! Would love to get this done
